zkevm-circuits/src/evm_circuit/execution/create.rs

```diff
@@ -0,0 +1,864 @@
+            ),
+        ] {
+            cb.call_context_lookup(true.expr(), None, field_tag, value);
+        }
+
+        let depth = cb.call_context(None, CallContextFieldTag::Depth);
+
+        for (field_tag, value) in [
+            (CallContextFieldTag::CallerId, cb.curr.state.call_id.expr()),
+            (CallContextFieldTag::IsSuccess, callee_is_success.expr()),
+            (
+                CallContextFieldTag::IsPersistent,
+                callee_reversion_info.is_persistent(),
+            ),
+            (CallContextFieldTag::TxId, tx_id.expr()),
+            (
+                CallContextFieldTag::CallerAddress,
+                from_bytes::expr(&caller_address.cells),
+            ),
+            (CallContextFieldTag::CalleeAddress, new_address),
+            (
+                CallContextFieldTag::RwCounterEndOfReversion,
+                callee_reversion_info.rw_counter_end_of_reversion(),
+            ),
+            (CallContextFieldTag::Depth, depth.expr() + 1.expr()),
+            (CallContextFieldTag::IsRoot, false.expr()),
+            (CallContextFieldTag::IsStatic, false.expr()),
+            (CallContextFieldTag::IsCreate, true.expr()),
+            (CallContextFieldTag::CodeHash, code_hash.expr()),
+        ] {
+            cb.call_context_lookup(true.expr(), Some(callee_call_id.expr()), field_tag, value);
+        }
+
+        cb.condition(initialization_code.has_length(), |cb| {
+            cb.require_step_state_transition(StepStateTransition {
+                rw_counter: Delta(cb.rw_counter_offset()),
+                call_id: To(callee_call_id.expr()),
+                is_root: To(false.expr()),
+                is_create: To(true.expr()),
+                code_hash: To(code_hash.expr()),
+                gas_left: To(callee_gas_left),
+                reversible_write_counter: To(3.expr()),
+                ..StepStateTransition::new_context()
+            })
+        });
+
+        cb.condition(not::expr(initialization_code.has_length()), |cb| {
+            cb.require_step_state_transition(StepStateTransition {
+                rw_counter: Delta(cb.rw_counter_offset()),
+                program_counter: Delta(1.expr()),
+                stack_pointer: Delta(2.expr() + is_create2.expr()),
+                gas_left: Delta(-gas_cost),
+                reversible_write_counter: Delta(5.expr()),
+                ..Default::default()
+            })
+        });
+
+        let keccak_input = cb.query_cell();
+        let keccak_input_length = cb.query_cell();
+        cb.condition(is_create2.expr(), |cb| {
+            // For CREATE2, the keccak input is the concatenation of 0xff, address, salt,
+            // and code_hash. Each sequence of bytes occurs in a fixed position, so to
+            // compute the RLC of the input, we only need to compute some fixed powers of
+            // the randomness.
+            let randomness_raised_to_16 = cb.power_of_randomness()[15].clone();
```

Comment:
it seems the rlc here assumes randomness of evm_word be equal to randomness of keccak_input?  it will be incorrect after the evm circuit challenge api refactor